### PR TITLE
#13782: Update .clang-tidy

### PR DESCRIPTION
**Pull request by blozano-tt**
### Ticket
#13782 

### Problem description
`.clang-tidy` was too generic

### What's changed
Stole settings from @dmakoviichuk-tt 

### Checklist
- [ ] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: .clang-tidy [hunk 1]
```diff
@@ -1,10 +1,24 @@
-Checks: >
-  bugprone-*,
-  performance-*,
-  modernize-*,
-  readability-*,
-  cppcoreguidelines-*
-  -modernize-use-trailing-return-type
+Checks: "*,
```
Comment (patrickroberts) [inline .clang-tidy:1]: Enabling everything and then opting out of a bunch of categories seems very weird to me. This implies that if new categories of checks get added to clang-tidy, when we upgrade, we'll opt into them without any changes to this file. Am I missing something?

File: .clang-tidy [hunk 2]
```diff
@@ -1,10 +1,24 @@
...
+        -google-*,
+        -llvm*,
+        -modernize-use-trailing-return-type,
+        -zircon-*,
+        -readability-else-after-return,
+        -readability-static-accessed-through-instance,
+        -readability-avoid-const-params-in-decls,
+        -cppcoreguidelines-non-private-member-variables-in-classes,
+        -misc-non-private-member-variables-in-classes,
+        -include-what-you-use,
```
Comment (blozano-tt) [inline .clang-tidy:15]: why disable include what you use?

File: .clang-tidy (hunk 1 above)
Comment (blozano-tt) [reply .clang-tidy:1]: @patrickroberts apparently json uses the same style

https://github.com/nlohmann/json/blob/develop/.clang-tidy

Comment (afuller-TT) [reply .clang-tidy:1]: @patrickroberts that's exactly what it means.  And IMO it's a feature, not a bug.

If we opt-in to checks, then we will miss new ones that are added (until we do considerable legwork).  Provided we have a fixed version of clang-tidy that we're using in CI/CD so upgrades aren't "magical", then we're stable and when we DO upgrade we immediately gain the new checks that we haven't yet violated, and we are told immediately which new checks we're violating.  We can then add the latter as disabled and we've gained new checks for essentially free, and have a clear TODO list that we can prioritize.  win-win.

Comment (afuller-TT) [reply .clang-tidy:1]: @blozano-tt with the current set, are we able to scan the entire repo and come up clean?

File: .clang-tidy (hunk 2 above)
Comment (afuller-TT) [reply .clang-tidy:15]: You got me excited that there is such a check.  I don't see it: https://clang.llvm.org/extra/clang-tidy/checks/list.html
Though they did add some header checks which will be useful for us.  But I was really hoping for one that'll tell us when we can get away with a forward declaration, which I'm not seeing.

Comment (afuller-TT) [reply .clang-tidy:15]: I assume this list isn't vetted for our repo.  Let's add a comment so when we have time to scan our repo and find the proper ignore list we'll know to do so and not trust this list here.

File: .clang-tidy (hunk 1 above)
Comment (blozano-tt) [reply .clang-tidy:1]: @afuller-TT I haven't tried, but I doubt it.

**Comment by blozano-tt** [comment]
Closing this PR, because it is used by clang-tidy bot.

To keep PRs less noisy, I want us to only enable singular checks that we all know and love.

**Pull request closed without merging**